Remotion/ObjectBinding/Web/UI/Controls/BocTreeNode.cs

```diff
@@ -163,47 +171,52 @@ private void EnsureBusinessObject()
     }
     else
     {
-      IBusinessObjectReferenceProperty property = Property;
+      IBusinessObjectReferenceProperty? property = Property;
       string businessObjectID = ItemID;
       BusinessObject = ((IBusinessObjectClassWithIdentity) property.ReferenceClass).GetObject (businessObjectID);
     }
   }
 
+  [MemberNotNull (nameof (_property))]
   private void EnsureProperty()
   {
     if (_property != null)
       return;
 
-    BusinessObjectTreeNode businessObjectParentNode = ParentNode as BusinessObjectTreeNode;
-    BusinessObjectPropertyTreeNode propertyParentNode = ParentNode as BusinessObjectPropertyTreeNode;
+    BusinessObjectTreeNode? businessObjectParentNode = ParentNode as BusinessObjectTreeNode;
+    BusinessObjectPropertyTreeNode? propertyParentNode = ParentNode as BusinessObjectPropertyTreeNode;
     
     if (businessObjectParentNode != null)
     {
-      IBusinessObjectProperty property = 
+      Assertion.IsNotNull (businessObjectParentNode.BusinessObject, "businessObjectParentNode.BusinessObject is not null.");
+      Assertion.IsNotNull (_propertyIdentifier, "_propertyIdentifier must not be null.");
+
+      IBusinessObjectProperty? property = 
           businessObjectParentNode.BusinessObject.BusinessObjectClass.GetPropertyDefinition (_propertyIdentifier);
-      Property = (IBusinessObjectReferenceProperty) property;
+      Property = (IBusinessObjectReferenceProperty?) property;
 
       if (_property == null) // This test could be omitted if graceful recovery is wanted.
         throw new InvalidOperationException ("Could not find IBusinessObjectReferenceProperty '" + _propertyIdentifier + "'.");
     }
     else if (propertyParentNode != null)
     {
       Property = propertyParentNode.Property;
-      return;
     }
+
+    Assertion.DebugIsNotNull (_property, "_property must not be null at this point.");
```

Comment:
Might trigger a false positive check. Needs a todo-ticket to check the correct/allowed flows in this method. 

EnsureProperty can currently not ensure that the property will never be null.